src/components/BuyItheumModal.tsx

```diff
@@ -164,6 +166,15 @@ const TransactionModal: React.FC<TransactionModalProps> = ({ isOpen, onClose, ad
             console.log("Transaction successful!");
             handleClose();
             clearInterval(intervalId);
+            (async () => {
+              const _token = await getAccountTokenFromApi(
+                address,
+                contractsForChain(networkConfiguration.mainnet.chainID).itheumToken,
+                networkConfiguration.mainnet.chainID
+              );
+              const balance = _token ? convertWeiToEsdt(_token.balance, _token.decimals).toNumber() : 0;
+              updateItheumBalance(balance);
```

Comment:
https://github.com/Itheum/data-dex/blob/main/src/store/StoreProvider.tsx#L186

This needs to update the "updateItheumBalance" store item.

The store method can be imported like this:
https://github.com/Itheum/data-dex/blob/main/src/store/StoreProvider.tsx#L33